src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2346,9 +2346,16 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 
 	MonoClass *klass = cmethod->klass;
 	MonoType *etype = mono_class_get_context (klass)->class_inst->type_argv [0];
-	gboolean supported = TRUE;
+	gboolean is_primitive_element_type = MONO_TYPE_IS_VECTOR_PRIMITIVE (etype);
 
-	if (!MONO_TYPE_IS_VECTOR_PRIMITIVE (etype))
```

Comment:
Same change should be done to `emit_sys_numerics_vector_t` as well.

Reply:
I have reverted the initial change and made another attempt to properly fix this.

Regarding `emit_sys_numerics_vector_t` and `emit_vector256_t` those functions are actually doing the right thing:
1. Firstly, filtering out non-primitive types with `MONO_TYPE_IS_VECTOR_PRIMITIVE`
2. Second, returning `TRUE` for `SN_get_IsSupported` intrinsic


- `emit_sys_numerics_vector_t`: 
https://github.com/dotnet/runtime/blob/6c38f2ceafa32fae0d775f862fb0fe43af5c200d/src/mono/mono/mini/simd-intrinsics.c#L3103-L3121

- `emit_vector256_t`:
https://github.com/dotnet/runtime/blob/6c38f2ceafa32fae0d775f862fb0fe43af5c200d/src/mono/mono/mini/simd-intrinsics.c#L5284-L5301